### PR TITLE
Mrtk Raystep Fix

### PR DESCRIPTION
**Pull request by Alexees**
- Fixes: #4080 .

### CODE REVIEW

**Comment by Railboy** [comment]
Thank you this was bugging me.

File: Assets/MixedRealityToolkit/Definitions/Physics/RayStep.cs [hunk 1]
```diff
@@ -155,33 +155,33 @@ public static Vector3 GetPointByDistance(RayStep[] steps, float distance)
...
+            Debug.Assert(steps != null && steps.Length > 0);
 
-            int numSteps = steps.Length;
+            float traveledDistance = 0;
             float stepLength = 0;
+            RayStep currentStep = default;
+
 
-            for (int i = 0; i < numSteps; i++)
+            foreach (var step in steps)
```
Comment (Railboy) [inline Assets/MixedRealityToolkit/Definitions/Physics/RayStep.cs]: Looks good. Works as expected.

**Comment by julenka** [comment]
azp /run mrtk_pr

**Comment by julenka** [comment]
Thank you for the fix! Next time, it would be great if you could explain in the PR the bug that you are fixing and the changes you made so we don't need to click through to the bug.

**Comment by Alexees** [comment]
@julenka I thought it was sufficient to link the issue, which explains the bug. I guess instead of commenting the issue with what's actually going on, I put that into the PR next time.

**Comment by julenka** [comment]
Hey @Alexees actually, looking more at what e.g. @luis-valverde-ms  is doing, it looks like we are following patterns that we do just refer to bugs, if the bugs are detailed enough. So, no worries, I think what you did here was fine.

**Pull request merged**